experimental/unitsconversion/src/converter.rs

```diff
@@ -158,4 +177,176 @@ impl<'data> ConverterFactory<'data> {
             Convertibility::NotConvertible
         }
     }
+
+    fn apply_si_prefix(si_prefix: &SiPrefix, ratio: &mut Ratio<BigInt>) {
+        match si_prefix.base {
+            Base::Decimal => {
+                *ratio *= Ratio::<BigInt>::from_integer(10.into()).pow(si_prefix.power as i32);
+            }
+            Base::Binary => {
+                *ratio *= Ratio::<BigInt>::from_integer(2.into()).pow(si_prefix.power as i32);
+            }
+        }
+    }
+
+    fn apply_power(power: i8, ratio: &mut Ratio<BigInt>) {
+        *ratio = ratio.pow(power as i32);
+    }
+
+    fn add_term(
+        &self,
+        unit_item: &MeasureUnitItem,
+        sign: i8,
+        conversion_rate: &mut Ratio<BigInt>,
```

Comment:
this is a almost a pure output parameter; just return the ratio and let the caller do the multiplication.